src/main/java/io/weaviate/client/v1/async/users/api/common/AssignedRolesGetter.java

```diff
@@ -0,0 +1,56 @@
+package io.weaviate.client.v1.async.users.api.common;
+
+import java.util.List;
+import java.util.concurrent.Future;
+
+import org.apache.hc.client5.http.impl.async.CloseableHttpAsyncClient;
+import org.apache.hc.core5.concurrent.FutureCallback;
+
+import io.weaviate.client.Config;
+import io.weaviate.client.base.AsyncBaseClient;
+import io.weaviate.client.base.AsyncClientResult;
+import io.weaviate.client.base.Result;
+import io.weaviate.client.v1.auth.provider.AccessTokenProvider;
+import io.weaviate.client.v1.rbac.api.WeaviateRole;
+import io.weaviate.client.v1.rbac.model.Role;
+
+public class AssignedRolesGetter extends AsyncBaseClient<List<Role>> implements AsyncClientResult<List<Role>> {
+  private String userId;
+  private boolean includePermissions = false;
+
+  private final String userType;
+
+  public AssignedRolesGetter(CloseableHttpAsyncClient httpClient, Config config, AccessTokenProvider tokenProvider,
+      String userType) {
+    super(httpClient, config, tokenProvider);
+    this.userType = userType;
+  }
+
+  public AssignedRolesGetter withUserId(String userId) {
+    this.userId = userId;
+    return this;
+  }
+
+  /**
+   * Include a full list of permissions for each role.
+   * If not set, only role names will be populated.
+   */
+  public AssignedRolesGetter includePermissions() {
+    return includePermissions(true);
+  }
+
+  public AssignedRolesGetter includePermissions(boolean include) {
+    this.includePermissions = include;
+    return this;
+  }
+
+  @Override
+  public Future<Result<List<Role>>> run(FutureCallback<Result<List<Role>>> callback) {
+    return sendGetRequest(path(), callback, Result.arrayToListParser(WeaviateRole[].class, WeaviateRole::toRole));
+  }
+
+  private String path() {
+    return String.format("/authz/users/%s/roles/%s?includeFullRoles=%s",
```

Comment:
question: setting with a `includePermissions` is used to control `includeFullRoles` param yes? shouldn't it be called `includeFullRoles` instead? or this is the name that we all agreed to use

Reply:
yes, I think so -- I borrowed the name from [the Python's implementation](https://github.com/weaviate/weaviate-python-client/blob/229b4af655c5f0b47ff00979e0a2f7904e83c781/weaviate/users/executor.py#L45) and used it in TS and Java